managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/EntryCacheManager.java

```diff
@@ -45,9 +47,12 @@
 @SuppressWarnings("checkstyle:javadoctype")
 public class EntryCacheManager {
 
-    private final long maxSize;
-    private final long evictionTriggerThreshold;
-    private final double cacheEvictionWatermark;
+    @Setter
+    private volatile long maxSize;
+    @Setter
```

Comment:
See my comments below
We should add a new method and do not expose internals otherwise we are breaking encapsulation

Reply:
Fixed